src/firebase/subscriber.js

```diff
@@ -0,0 +1,59 @@
+import firebase from './firebase';
+
+const subscribe = () => {
+  // get ref to custom storage bucket
+  const topicName = process.env.TOPIC_NAME;
+  const storageRef = firebase.app()
+    .storage(process.env.STORAGE_PROCESSED_IMAGES_BUCKET).ref();
+
+  // subscribe to the pub/sub topic for image process status
+  firebase.functions.pubsub.topic(topicName).onPublish((message) => {
+    const { userId, experimentId, resultImages } =
+      handleMessage(message, storageRef);
+    if (resultImages.length > 0) {
+      // store into firestore.
+      const documentId = userId + '-' + experimentId;
+      firebase.firestore().collection('Images')
+        .doc(documentId).set(resultImages);
+    }
+  });
+};
+
+/**
+ * listen to the Pub/Sub topic message queue and collect image urls.
+ * @param {String} message image process result messages.
+ * @param {*} bucketRef processed images bucket in cloud storage.
+ * @return {String[]} array of processed image urls.
+ */
+const handleMessage = (message, bucketRef) => {
+  // Decode the PubSub Message body.
+  const messageBody = message.data ?
+    Buffer.from(message.data, 'base64').toString() : null;
+  if (!messageBody) {
+    return [];
+  }
+
+  const [userId, experimentId, status] = messageBody.split('-');
+  const resultImages = [];
+
+  if (status === 'complete') {
+    // get all processed images from cloud storage
+    const folderPath = userId + '-' + experimentId + '/';
+    const processedImagesRef = bucketRef.child(folderPath);
+
+    // store the download url in the database
+    processedImagesRef.getFiles((err, images) => {
+      if (!err) {
+        // files is an array of File objects.
+        images.forEach((image) => {
+          image.getDownloadURL().then((url) => {
+            resultImages.push(url);
```

Comment:
could you please make a TODO and another JIRA task for this? I don't think sending to frontend all 400 images is a good idea :/ how should we do this? @cmutuc25

Reply:
I think what we do now is storing the list of processed image URLs in the firestore images collection, the front end can decide how to deal with those URLs. We are not sending all these links directly to the frontend.

Reply:
oh whoops you are returning a list of urls my bad! It's all good then. Do you mind changing the variable name to resImgURLs or something?

Reply:
Sure! Let me push the changes shortly